app/js/main.js

```diff
@@ -201,6 +201,52 @@ function setupMeteringDemo(subscriptions) {
     location.reload();
   });
 
+  // Determine whether there is a publisher based entitlement
+  let entitlement = null;
+  let isUserRegistered = false; // Set this if known
+  let unlockContent = true;
+  // The publisher should modify this check so it accurately returns
+  // true if the publisher knows about the user's subscription
+  if (getQueryParams().hasPublisherSubscription === 'true') {
+    isUserRegistered = true;
+    entitlement = 'EVENT_SHOWCASE_UNLOCKED_BY_SUBSCRIPTION';
+    return;
+    // The publisher should modify this check so it accurately returns
+    // true if a publisher meter was used to unlock this article.
+  } else if (getQueryParams().consumedPublisherMeter === 'true') {
+    isUserRegistered = true;
+    entitlement = 'EVENT_SHOWCASE_UNLOCKED_BY_METER';
+    // The publisher should modify this check so it accurately returns
+    // true if the article is free (for either all or just registered users)
+  } else if (getQueryParams().isFreeArticle === 'true') {
+    // set isUserRegistered if known
+    entitlement = 'EVENT_SHOWCASE_UNLOCKED_FREE_PAGE';
+  } else {
+    // If the publisher is not unlocking the article, they should
+    // let google know if the user is registered.  Google will
+    // generate a showcase entitlement if one of our meters is
+    // consumed or the user owns a subscription through Google.
+    // Google will also automatically generate the regwall event,
+    // so the publisher can choose to report this for registered
+    // users only if they prefer.
+    unlockContent = false;
+    entitlement = isUserRegistered
+      ? 'EVENT_SHOWCASE_NO_ENTITLEMENTS_PAYWALL'
+      : 'EVENT_SHOWCASE_NO_ENTITLEMENTS_REGWALL';
+    subscriptions.setShowcaseEntitlement({isUserRegistered, entitlement});
+  }
+
+  // Inform showcase of the  publisher's entitlement decision.  If google
+  // decides to unlock the article, an additional event indicating why
+  // will automatically be generated.
+  subscriptions.setShowcaseEntitlement({isUserRegistered, entitlement});
+
+  // If we unlocked the content, display it and exit.
+  if (unlockContent) {
```

Comment:
The two cases where `unlockContent` is true here could actually be their own flows. I think this would make the code clearer

Reply:
If those cases were their own flows, then we wouldn't need special URL parameters for them. Instead we could just specify the flow like "?eventShowcaseUnlockedByMeter"